### PR TITLE
Apple Pay Integration [Unit Tests]

### PR DESCRIPTION
**Pull request by mpetrenco**
Added unit tests for ApplePayConfiguration and ApplePayManager

### CODE REVIEW

File: Tests/APITests/ApplePayTests.swift [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+class ApplePayTests: JudoTestCase {
+    
+    var configuration: ApplePayConfiguration!
+    var manager: ApplePayManager!
+    
+    override func setUp() {
+        super.setUp()
+        
+        let items = [
+            PaymentSummaryItem(label: "Item 1", amount: NSDecimalNumber(string: "0.001$")),
```
Comment (zfoltin) [inline Tests/APITests/ApplePayTests.swift]: to be more realistic, use 2 decimal places. Also remove the `$`

File: Tests/APITests/ApplePayTests.swift [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    func test_OnApplePayConfigurationInitialization_DefaultValuesAreRespected() {
+        XCTAssertEqual(configuration.merchantCapabilities, MerchantCapability.capability3DS,
+                       "Configuration's merchantCapabilities does not default to 3D Security")
+        XCTAssertTrue(configuration.supportedCardNetworks.contains(NSNumber(value: CardNetwork.visa.rawValue)),
+                       "Configuration's supportedCardNetworks does not contain Visa")
+        XCTAssertTrue(configuration.supportedCardNetworks.contains(NSNumber(value: CardNetwork.AMEX.rawValue)),
+                      "Configuration's supportedCardNetworks does not contain AmEx")
+        XCTAssertTrue(configuration.supportedCardNetworks.contains(NSNumber(value: CardNetwork.masterCard.rawValue)),
+                      "Configuration's supportedCardNetworks does not contain MasterCard")
+        XCTAssertTrue(configuration.supportedCardNetworks.contains(NSNumber(value: CardNetwork.maestro.rawValue)),
```
Comment (zfoltin) [inline Tests/APITests/ApplePayTests.swift]: only do this check on iOS 12+

File: Tests/APITests/ApplePayTests.swift [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    }
+    
+    /**
+     * GIVEN:  ApplePayConfiguration is initialized correctly
+     *
+     *  THEN:  ApplePayManager should return a PKAuthorizationViewController object when calling
+     *         its "pkPaymentAuthorizationViewController" instance method
+     */
+    func test_OnValidConfiguration_ManagerReturnsValidPKAuthorizationViewController() {
+        XCTAssertNotNil(manager.pkPaymentAuthorizationViewController(),
```
Comment (zfoltin) [inline Tests/APITests/ApplePayTests.swift]: might worth asserting the type of the returned VC - as it says in the test name and comment

File: .gitmodules [hunk 4]
```diff
@@ -4,7 +4,3 @@
 [submodule "TrustKit"]
 	path = TrustKit
 	url = https://github.com/datatheorem/TrustKit.git
-[submodule "ZappMerchantLib"]
```
Comment (zfoltin) [inline .gitmodules]: Something must have gone wrong with merging in develop (maybe that this PR is based off of the `feature/apple-pay-demo` branch. Changes to `.gitmodules`, `podspec` and others shouldn't be part of this PR

Comment (mpetrenco) [reply .gitmodules]: Right. Sorry about that.

File: Tests/APITests/ApplePayTests.swift [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    /**
+     * GIVEN:  ApplePayConfiguration is initialized correctly
+     *
+     *  THEN:  ApplePayManager should return a valid JPAmount object when calling
+     *         its "jpAmount" instance method
+     */
+    func test_OnValidConfiguration_ManagerGeneratesCorrectJPAmount() {
+        XCTAssertNotNil(manager.jpAmount(),
+                        "JPAmount must not be nil on correct configuration")
+        XCTAssertEqual(manager.jpAmount().amount, "0.003",
```
Comment (zfoltin) [inline Tests/APITests/ApplePayTests.swift]: I guess this should break now, no?

**Comment by zfoltin** [comment]
@mpetrenco  Could you revert the git submodule changes in this PR please?
